lib/cli/src/commands/run.rs

```diff
@@ -575,7 +576,28 @@ impl ExecutableTarget {
                 let wasm = std::fs::read(path)?;
                 let engine = runtime.engine().context("No engine available")?;
                 pb.set_message("Compiling to WebAssembly");
-                let module = Module::new(&engine, wasm)?;
+
+                let tasks = runtime.task_manager();
+                let module_cache = runtime.module_cache();
+                let module_hash = ModuleHash::sha256(&wasm);
+
+                let module = match tasks.block_on(module_cache.load(module_hash, &engine)) {
```

Comment:
Maybe leave a TODO/followup issue that this should switch to `Runtime::load_module` once #3996 is merged?

Reply:
I've added a comment on #3996 so hopefully that PR will be able to do the fix immediately.